src/main/java/com/minecolonies/core/colony/CitizenData.java

```diff
@@ -916,9 +917,36 @@ public void updateEntityIfNecessary()
             }
         }
 
+        //Check if we are traveling, we don't spawn an entity if we are traveling.
+        if (getColony().getTravelingManager().isTravelling(this))
+        {
+            return;
+        }
+
+        boolean spawnVisible = true;
+
+        //Okey we are either just done traveling or the entity disappeared, lets check if we just finished traveling.
+        final Optional<BlockPos> travelingTargetCandidate = getColony().getTravelingManager().getTravellingTargetFor(this);
+        if (travelingTargetCandidate.isPresent())
+        {
+            //We just finished traveling, lets spawn the entity by setting the nextRespawnPosition.
+            getColony().getTravelingManager().finishTravellingFor(this);
+            nextRespawnPos = travelingTargetCandidate.get();
+            spawnVisible = false;
+            lastPosition = nextRespawnPos;
+        }
+
         if (nextRespawnPos != null)
         {
-            colony.getCitizenManager().spawnOrCreateCivilian(this, colony.getWorld(), nextRespawnPos, true);
+            ICitizenData data = colony.getCitizenManager().spawnOrCreateCivilian(this, colony.getWorld(), nextRespawnPos, true);
+            getEntity().get().getCitizenJobHandler().setModelDependingOnJob(data.getJob());
+            if (!spawnVisible)
+            {
+                data.getEntity().ifPresent(entity -> {
```

Comment:
we already do a getEntity().get() before. So maybe we could just cache the entity, != null and then use it directly